Open Judge System/OJS.Services/Data/ParticipantScores/ParticipantScoresDataService.cs

```diff
@@ -0,0 +1,119 @@
+﻿namespace OJS.Services.Data.ParticipantScores
+{
+    using System.Linq;
+    using System.Transactions;
+
+    using EntityFramework.Extensions;
+
+    using OJS.Data.Models;
+    using OJS.Data.Repositories.Contracts;
+    using OJS.Services.Data.Participants;
+
+    public class ParticipantScoresDataService : IParticipantScoresDataService
+    {
+        private readonly IEfGenericRepository<ParticipantScore> participantScores;
+        private readonly IParticipantsDataService participantsData;
+
+        public ParticipantScoresDataService(
+            IEfGenericRepository<ParticipantScore> participantScores,
+            IParticipantsDataService participantsData)
+        {
+            this.participantScores = participantScores;
+            this.participantsData = participantsData;
+        }
+
+        public ParticipantScore Get(int participantId, int problemId) =>
+            this.participantScores
+                .All()
+                .FirstOrDefault(ps => ps.ParticipantId == participantId &&
+                    ps.ProblemId == problemId);
+
+        public ParticipantScore Get(int participantId, int problemId, bool isOfficial) =>
+            this.participantScores
+                .All()
+                .FirstOrDefault(ps => ps.ParticipantId == participantId &&
+                    ps.ProblemId == problemId &&
+                    ps.IsOfficial == isOfficial);
+
+        public IQueryable<ParticipantScore> GetAll()
+            => this.participantScores.All();
+
+        public bool Save(Submission submission, bool resetScore = false)
+        {
+            using (var transaction = new TransactionScope())
```

Comment:
Most of the times this method is called, it's called _within_ a transaction. Are you sure that nesting transactions is a good idea? Have you considered what would happen if an error occurs in the inner transaction and how would it affect the outer one? [Here's some useful info](https://stackoverflow.com/a/2794855/5604114)